pkg/kubelet/cm/helpers_linux.go

```diff
@@ -44,3 +48,36 @@ func GetCgroupSubsystems() (*CgroupSubsystems, error) {
 		MountPoints: mountPoints,
 	}, nil
 }
+
+// readProcsFile takes a cgroup directory name as an argument
+// reads through the cgroup's procs file and returns a list of tgid's.
+// It returns an empty list if a procs file doesn't exists
+func readProcsFile(dir string) ([]int, error) {
+	procsFile := filepath.Join(dir, "cgroup.procs")
```

Comment:
Yes, I had verified that earlier. tasks includes all the thread ids. and procs file includes just the thread group ids. So when we do the following all
https://github.com/kubernetes/kubernetes/pull/29049/files#diff-ccf7ec0d731f99a27b208c6edfa89576R177

That should take care of all the processes attached to the cgroup.